### PR TITLE
Revised unAttachedDiskByokEncryptionEnabled

### PR DESCRIPTION
**Pull request by AkhtarAmir**

### CODE REVIEW

File: plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+var async = require('async');
+
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Unattached Disk Volumes with Default Encryption',
+    category: 'Virtual Machines',
+    domain: 'Compute',
+    severity: 'Medium',
+    description: 'Ensures that no default encrypted Azure virtual machine disks are in unattached state.',
```
Comment (fatima99s) [inline plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js:10]:
```suggestion
    description: 'Ensures that unattached Azure virtual machine disks have BYOK (Customer-Managed Key) encryption enabled.',
```

File: plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                return rcb();
+            }
+
+            for (let disk of disks.data) {
+                if (!disk.id) continue;
+
+                if (disk.encryption && disk.encryption.type &&
+                    disk.encryption.type === 'EncryptionAtRestWithPlatformKey' &&
+                    disk.diskState && disk.diskState.toLowerCase() === 'unattached') {
+                    helpers.addResult(results, 2, 'Disk volume is unattached and encrypted with default encryption key', location, disk.id);
```
Comment (fatima99s) [inline plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js:43]:
```suggestion
                    helpers.addResult(results, 2, ' Unattached disk volume has BYOK encryption disabled', location, disk.id);
```

File: plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                } else {
+                    helpers.addResult(results, 0, 'Disk volume is attached or encrypted with BYOK', location, disk.id);
```
Comment (fatima99s) [inline plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js:45]:
```suggestion
                    helpers.addResult(results, 0, 'Unattached disk volume has BYOK encryption enabled', location, disk.id);
```

File: plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+var async = require('async');
+
+var helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'Unattached Disk Volumes with Default Encryption',
```
Comment (fatima99s) [inline plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js:6]:
```suggestion
    title: 'Unattached Disk Volumes BYOK Encryption Enabled',
```

File: plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                return rcb();
+            }
+            if (!disks.data.length) {
+                helpers.addResult(results, 0, 'No existing VM disk volumes found', location);
+                return rcb();
+            }
+
+            for (let disk of disks.data) {
+                if (!disk.id) continue;
+
```
Comment (fatima99s) [inline plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js:39]: there should be a check for the disk state if its attached skip the result

File: plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+                helpers.addResult(results, 0, 'No existing VM disk volumes found', location);
+                return rcb();
+            }
+
+            for (let disk of disks.data) {
+                if (!disk.id) continue;
+                if (disk.diskState && disk.diskState.toLowerCase() === 'unattached') {
+                     if (disk.encryption && disk.encryption.type &&
+                         disk.encryption.type === 'EncryptionAtRestWithPlatformKey') {
+                         helpers.addResult(results, 2, 'Unattached disk volume has BYOK encryption disabled', location, disk.id);
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js:42]:
```suggestion
                        helpers.addResult(results, 2, 'Unattached disk volume has BYOK encryption disabled', location, disk.id);
```

File: plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+                     } else {
```
Comment (alphadev4) [inline plugins/azure/virtualmachines/unAttachedDiskByokEncryptionEnabled.js:43]:
```suggestion
                    } else {
```

**Review by alphadev4 — APPROVED** [review]

**Pull request merged**